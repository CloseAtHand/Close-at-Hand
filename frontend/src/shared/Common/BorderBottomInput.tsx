import { StyleSheet, TextInput } from "react-native";
import COLORS from "../../app/constant/COLORS";

interface Props {
	value: string;
	onChangeText: (text: string) => void;
	placeholder: string;
	secureTextEntry?: boolean;
}

const BorderBottomInput: React.FC<Props> = ({
	value,
	onChangeText,
	placeholder,
	secureTextEntry = false,
}) => {
	function handleChange(text: string) {
		onChangeText(text);
	}

	return (
		<TextInput
			secureTextEntry={secureTextEntry}
			style={styles.input}
			value={value}
			onChangeText={handleChange}
			placeholder={placeholder}
			testID="input"
		/>
	);
};

export default BorderBottomInput;

const styles = StyleSheet.create({
	input: {
		backgroundColor: COLORS.White,
		borderColor: COLORS.Black,
		borderBottomWidth: 1,

		fontSize: 16,
		textAlign: "left",

		height: 40,
		marginVertical: 10,
		marginHorizontal: 20,
		padding: 10,
		width: "auto",
	},
});
