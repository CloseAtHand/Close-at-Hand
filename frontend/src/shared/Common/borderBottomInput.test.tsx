import "@testing-library/jest-native/";
import { render, fireEvent } from "@testing-library/react-native";
import renderer from "react-test-renderer";

import BorderBottomInput from "./BorderBottomInput";

describe("BorderBottomInput", () => {
	const INITIAL_VALUE = "value";
	const TestID = "input";

	it("스냅샷 테스트", () => {
		const tree = renderer
			.create(
				<BorderBottomInput
					value={INITIAL_VALUE}
					onChangeText={jest.fn()}
					placeholder="플레이스홀더"
				/>,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("placeholder가 작성됩니다.", () => {
		const { getByPlaceholderText } = render(
			<BorderBottomInput
				value={INITIAL_VALUE}
				onChangeText={jest.fn()}
				placeholder="플레이스홀더"
			/>,
		);
		expect(getByPlaceholderText("플레이스홀더")).toBeTruthy();
	});

	it("value를 렌더링합니다.", () => {
		const { getByTestId } = render(
			<BorderBottomInput
				value={INITIAL_VALUE}
				onChangeText={jest.fn()}
				placeholder="플레이스홀더"
			/>,
		);
		expect(getByTestId(TestID)).toBeTruthy();
	});

	it("텍스트를 입력하면 setValue가 입력값을 받아 실행됩니다.", () => {
		const setValue = jest.fn();
		const { getByTestId } = render(
			<BorderBottomInput
				value={INITIAL_VALUE}
				onChangeText={setValue}
				placeholder="플레이스홀더"
			/>,
		);

		const input = getByTestId(TestID);

		fireEvent.changeText(input, "새 입력");

		expect(setValue).toHaveBeenCalledWith("새 입력");
	});
});
