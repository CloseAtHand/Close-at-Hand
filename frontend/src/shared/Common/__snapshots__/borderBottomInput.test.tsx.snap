// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BorderBottomInput 스냅샷 테스트 1`] = `
<TextInput
  onChangeText={[Function]}
  placeholder="플레이스홀더"
  style={
    {
      "backgroundColor": "#ffffff",
      "borderBottomWidth": 1,
      "borderColor": "#121212",
      "fontSize": 16,
      "height": 40,
      "marginHorizontal": 20,
      "marginVertical": 10,
      "padding": 10,
      "textAlign": "left",
      "width": "auto",
    }
  }
  testID="input"
  value="value"
/>
`;
